### DISCUSSION TITLE
add other keys in select selected components (in a form)

### DISCUSSION THREAD

**Discussion by Ladvace** [ant-design/ant-design#48980] [Q&A] [answered]
is it possible to add other keys when selecting an element in a select component other than `key` and `value` (and `key`)?

**Comment by wanpan11** [comment]
You can use `fieldNames` to convert.

![image](https://github.com/ant-design/ant-design/assets/38714194/8a52fb43-0c61-4b3d-ad7e-4f8147ebc597)

**Comment by Ladvace** [comment]
can you use it convert it to a custom object?

**Comment by wanpan11** [comment]
> can you use it convert it to a custom object?

This should work
```
 <Select
      value="1"
      fieldNames={{ value: 'name', label: 'title' }}
      options={[
        { name: '1', title: 'Jack', source: 1 },
        { name: '2', title: 'Lucy', source: 2 },
        { name: '3', title: 'yiminghe', source: 3 },
        { name: '4', title: 'Disabled', disabled: true, source: 4 },
      ]}
      onChange={(_, item) => {
        console.log(item);
      }}
    />
```

![image](https://github.com/ant-design/ant-design/assets/38714194/22467822-015b-4d75-98af-2270bd58e8ca)

**Reply by Ladvace** [reply]
I'm gonna test it out, thanks!

**Reply by Ladvace** [reply]
I forgot to mention that I'm using it in a form, this doesn't solve the problem, I just want to add more keys, this works fine if you use the onChange but not if this is in a form using the onFinish

**Comment by wanpan11** [comment, marked as answer]
This example should meet your needs
https://stackblitz.com/edit/react-jyqcpj-pyfzkx?file=demo.tsx

![image](https://github.com/ant-design/ant-design/assets/38714194/1f09e1d5-d783-4791-b94a-975dce1d7f19)

**Reply by Ladvace** [reply]
this works! another question, what about if you want that in a deselect?

**Reply by wanpan11** [reply]
> this works! another question, what about if you want that in a deselect?

It can be used like a normal Select component, only the props of MySelect are special.

![image](https://github.com/ant-design/ant-design/assets/38714194/f002a757-6327-4cc7-b14f-fd077085c82c)
